src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -984,13 +1047,24 @@ CreateHashDistributedTable(Oid relationId, char *distributionColumnName,
 	Var *distributionColumn = NULL;
 	Oid distributionColumnType = 0;
 	uint32 colocationId = INVALID_COLOCATION_ID;
+	char replicationModel = 0;
 
 	/* get distribution column type */
 	distributedRelation = relation_open(relationId, AccessShareLock);
 	distributionColumn = BuildDistributionKeyFromColumnName(distributedRelation,
 															distributionColumnName);
 	distributionColumnType = distributionColumn->vartype;
 
+	/* all hash-distributed tables with repfactor=1 are treated as MX tables */
+	if (replicationFactor == 1)
```

Comment:
Shouldn't we need to check `DISTRIBUTE_BY_HASH` here? Or maybe  shouldn't this need to passed from the caller function?